packages/smooth_app/lib/pages/scan/ml_kit_scan_page.dart

```diff
@@ -158,21 +145,59 @@ class MLKitScannerPageState extends State<MLKitScannerPage> {
         debugPrint(e.toString());
       }
     }
+
     if (mounted) {
       setState(() {});
     }
   }
 
+  void cameraListener() {
+    if (mounted) {
+      setState(() {});
+
+      if (_controller?.value.hasError == true) {
+        // TODO(M123): Handle errors better
+        debugPrint(_controller!.value.errorDescription);
+      }
+    }
+  }
+
   Future<void> _stopImageStream() async {
+    if (stoppingCamera) {
+      return;
+    }
+
     stoppingCamera = true;
     if (mounted) {
       setState(() {});
     }
+
+    _controller?.removeListener(cameraListener);
     await _controller?.dispose();
-    barcodeScanner?.close();
+    await barcodeScanner?.close();
+
     barcodeScanner = null;
-    _controller?.removeListener(() {});
     _controller = null;
+
+    if (mounted && ScreenVisibilityDetector.visible(context)) {
+      final DateTime referentialTime = DateTime.now();
+
+      WidgetsBinding.instance?.addPostFrameCallback((_) {
+        final int diff =
+            DateTime.now().difference(referentialTime).inMilliseconds;
```

Comment:
I currently don't understand why we use a PostFrameCallback here, can you explain your thinking behind this

Reply:
Basically, all the trick to detect the issue is here.

By invoking a post-frame callback, we have two scenarios:
- the screen is still on -> the callback will be called in the next frame (< 16 ms on a 60Hz display)
- the user leaves the app, changes the tab… -> this same callback will be called but only when resumed. In that case, the difference is more than a few frames